### PR TITLE
[WIP] Start unit testing report feature

### LINKED ISSUE #123: Write tests for the report API
We need some API tests for the report mutation.

### PR DESCRIPTION
**Pull request by mattwr18**
fixes #123 

would love some feedback on other tests I could be writing @appinteractive @roschaefer 
I'm sure you would be interested in having a look as well @Wolle

### CODE REVIEW

File: src/graphql-schema.spec.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
-import { request } from 'graphql-request'
+import { GraphQLClient, request } from 'graphql-request'
 import { create, cleanDatabase } from './seed/factories'
```
Comment (roschaefer) [inline src/graphql-schema.spec.js]:
```suggestion
import Factory from './seed/factories'
```

File: src/graphql-schema.spec.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
 import jwt from 'jsonwebtoken'
-import { host } from './jest/helpers'
+import { host, login } from './jest/helpers'
```
Comment (roschaefer) [inline src/graphql-schema.spec.js]:
```suggestion
const factory = Factory()
```

File: src/graphql-schema.spec.js [hunk 3]
```diff
@@ -60,3 +60,60 @@ describe('login', () => {
     })
   })
 })
+
+describe('report', () => {
+  beforeEach(async () => {
+    await create('user', {
```
Comment (roschaefer) [inline src/graphql-schema.spec.js]:
```suggestion
    await factory.create('user', {
```

File: src/graphql-schema.spec.js [hunk 4]
```diff
@@ -60,3 +60,60 @@ describe('login', () => {
...
+      email: 'test@example.org',
+      password: '1234'
+    })
+  })
+
+  afterEach(async () => {
+    await cleanDatabase()
```
Comment (roschaefer) [inline src/graphql-schema.spec.js]:
```suggestion
    await factory.cleanDatabase()
```

File: src/graphql-schema.spec.js [hunk 5]
```diff
@@ -60,3 +60,60 @@ describe('login', () => {
...
+  describe('unauthenticated', () => {
+    let client
+    it('throws authorization error', async () => {
+      client = new GraphQLClient(host)
+      await expect(
+        client.request(`mutation {
+        report(
+          description: "I don't like this user",
+          resource: {
+            id: "u1",
```
Comment (roschaefer) [inline src/graphql-schema.spec.js]: @mattwr18 who is user with id `u1`?

**Comment by mattwr18** [comment]
it was failing cause of linting, I have followed your `PR` review suggestions, although it required borrowing quite a bit of code from your work...

I also created an abusive-user with id u2 to report

**Comment by roschaefer** [comment]
@mattwr18 you scored first blood on our current burndown chart.

**Pull request merged**